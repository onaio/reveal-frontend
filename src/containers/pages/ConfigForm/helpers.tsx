import React from 'react';
import Loading from '../../../components/page/Loading';
import { NoDataComponent } from '../../../components/Table/NoDataComponent';
import { OPENSRP_API_BASE_URL } from '../../../configs/env';
import { getAcessTokenOrRedirect, growl } from '../../../helpers/utils';
import { newGetPayloadOptions } from '../../../services/opensrp';

/** drill down props */
export const drillDownProps = {
  paginate: false,
  renderNullDataComponent: () => <NoDataComponent />,
};

export const defaultConfigProps = {
  LoadingComponent: <Loading />,
  accessToken: getAcessTokenOrRedirect,
  baseURL: OPENSRP_API_BASE_URL,
  customAlert: growl,
  getPayload: newGetPayloadOptions,
};
