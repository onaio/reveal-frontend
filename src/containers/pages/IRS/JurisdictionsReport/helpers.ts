import { CellInfo, Column } from 'react-table';
import { getIRSThresholdAdherenceIndicator } from '../../../../helpers/indicators';
import { FlexObject } from '../../../../helpers/utils';

/** columns for Namibia IRS jurisdictions */
export const NamibiaColumns = [
  {
    Header: 'Name',
    columns: [
      {
        Header: '',
        accessor: 'jurisdiction_name',
        minWidth: 180,
      },
    ],
  },
  {
    Header: 'Structures Targeted',
    columns: [
      {
        Header: '',
        accessor: 'target_2019',
      },
    ],
  },
  {
    Header: 'Structures Found',
    columns: [
      {
        Header: '',
        accessor: 'structuresfound',
      },
    ],
  },
  {
    Header: 'Structures Sprayed',
    columns: [
      {
        Header: '',
        accessor: 'structuressprayed',
      },
    ],
  },
  {
    Header: 'Target Coverage',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'targetcoverage',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Found Coverage',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'foundcoverage',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Refusals',
    columns: [
      {
        Header: 'Following first visit',
        accessor: 'refusalsfirst',
        className: 'centered',
        id: 'refusalsfirst',
      },
      {
        Header: 'Following mop-up',
        accessor: 'refusalsmopup',
        className: 'centered',
        id: 'refusalsmopup',
      },
    ],
    headerClassName: 'centered',
  },
  {
    Header: 'Locked',
    columns: [
      {
        Header: 'Following first visit',
        accessor: 'lockedfirst',
        className: 'centered',
        id: 'lockedfirst',
      },
      {
        Header: 'Following mop-up',
        accessor: 'lockedmopup',
        className: 'centered',
        id: 'lockedmopup',
      },
    ],
    headerClassName: 'centered',
  },
];

/** columns for Zambia IRS jurisdictions */
export const ZambiaJurisdictionsColumns = [
  {
    Header: 'Name',
    columns: [
      {
        Header: '',
        accessor: 'jurisdiction_name',
        minWidth: 180,
      },
    ],
  },
  {
    Header: 'Total Spray Areas',
    columns: [
      {
        Header: '',
        accessor: 'totareas',
      },
    ],
  },
  {
    Header: 'Targeted Spray Areas',
    columns: [
      {
        Header: '',
        accessor: 'targareas',
      },
    ],
  },
  {
    Header: '% visited Spray Areas Effectively sprayed',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'perctvisareaseffect',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Total Structures',
    columns: [
      {
        Header: '',
        accessor: 'totstruct',
      },
    ],
  },
  {
    Header: 'Targeted Structures',
    columns: [
      {
        Header: '',
        accessor: 'targstruct',
      },
    ],
  },
  {
    Header: 'Sprayed Structures',
    columns: [
      {
        Header: '',
        accessor: 'sprayedstruct',
      },
    ],
  },
  {
    Header: 'Spray coverage of targeted (Progress)',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraycovtarg',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Room coverage of structures sprayed',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'roomcov',
        className: 'indicator centered',
      },
    ],
  },
];

/** columns for Namibia IRS focus (spray) areas */
export const ZambiaFocusAreasColumns = [
  {
    Header: 'Name',
    columns: [
      {
        Header: '',
        accessor: 'jurisdiction_name',
        minWidth: 180,
      },
    ],
  },
  {
    Header: 'Structures on the ground',
    columns: [
      {
        Header: '',
        accessor: 'totstruct',
      },
    ],
  },
  {
    Header: 'Found',
    columns: [
      {
        Header: '',
        accessor: 'foundstruct',
      },
    ],
  },
  {
    Header: 'Sprayed',
    columns: [
      {
        Header: '',
        accessor: 'sprayedstruct',
      },
    ],
  },
  {
    Header: 'Spray Coverage (Effectiveness)',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraycov',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Found Coverage',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraytarg',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Spray Success Rate (PMI SC)',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraysuccess',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Room coverage of structures sprayed',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'roomcov',
        className: 'indicator centered',
      },
    ],
  },
];

/** IRS Table Columns
 * These are all the table columns for IRS that we know about.
 */
export const IRSTableColumns: { [key: string]: Column[] } = {
  namibia2019: NamibiaColumns,
  zambiaFocusArea2019: ZambiaFocusAreasColumns,
  zambiaJurisdictions2019: ZambiaJurisdictionsColumns,
};
