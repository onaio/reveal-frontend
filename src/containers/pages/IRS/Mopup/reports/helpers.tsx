import { DropDownCellProps } from '@onaio/drill-down-table';
import { Dictionary } from '@onaio/utils';
import { ceil, get } from 'lodash';
import React from 'react';
import { Link } from 'react-router-dom';
import { Cell } from 'react-table';
import { renderPercentage } from '../../../../../helpers/indicators';
import { GenericJurisdiction } from '../../../../../store/ducks/generic/jurisdictions';
import { friendlyDate } from '../../../../../store/ducks/opensrp/events/utils';

/** focus Areas mop-up reporting columns */
export const ZambiaIRSMopupFocusArea = [
  {
    Header: 'Health Facility',
    accessor: 'jurisdiction_name',
  },
  {
    Header: 'Spray Area',
    accessor: 'health_center_jurisdiction_name',
  },
  {
    Header: 'Structures on Ground',
    accessor: 'totstruct',
  },
  {
    Header: 'Visited Sprayed',
    accessor: 'sprayedstruct',
  },
  {
    Cell: ({ value }: Cell) => ceil(value),
    Header: 'Structures remaining to spray to reach 90% SE',
    accessor: 'structures_remaining_to_90_se',
  },
  { Cell: ({ value }: Cell) => ceil(value), Header: 'TLA Days', accessor: 'tla_days_to_90_se' },
  {
    Cell: (cell: Cell) => renderPercentage(cell),
    Header: 'Spray Effectiveness',
    accessor: 'spray_effectiveness',
  },
  { Cell: (cell: Cell) => renderPercentage(cell), Header: 'Found coverage', accessor: 'spraytarg' },
  { Cell: (cell: Cell) => renderPercentage(cell), Header: 'Spray coverage', accessor: 'spraycov' },
  {
    Cell: ({ value }: Cell) => friendlyDate(value),
    Header: 'Date of Last Visit',
    accessor: 'latest_spray_event_date',
  },
  {
    Cell: ({ value }: Cell) => friendlyDate(value),
    Header: 'Date of last Decision form',
    accessor: 'latest_sa_event_date',
  },
];

export const zambiaMopupJurisdiction = [
  {
    Header: 'Name',
    accessor: 'jurisdiction_name',
  },
  {
    Header: 'Health Centers to Mop-up',
    accessor: 'num_health_centers_below_90_se',
  },
  {
    Header: 'Spray Areas to to Mop-up',
    accessor: 'num_spray_areas_below_90_se',
  },
  {
    Header: 'Structures to spray or areas to react 90%',
    accessor: 'structures_remaining_to_90_se',
  },
  {
    Cell: ({ value }: Cell) => ceil(value),
    Header: 'TLA days needed for areas needed to reach 90%',
    accessor: 'tla_days_to_90_se',
  },
];

/** column definitions for IRS mopup reports */
const planTableColumnsLookup = {
  ZambiaIRSMopupFocusArea,
  zambiaMopupJurisdiction,
};

/**
 * gets columns to be used
 * @param {GenericJurisdiction[]} jurisdiction - jurisdiction data
 * @param {string} jurisdictionColumn  - The reporting jurisdiction columns
 * @param {string} focusAreaColumn  - Reporting focus area column
 * @param {string} focusAreaLevel - Jurisdiction depth of the lowest level jurisdictions
 * @param {string} jurisdictionId - jurisdiction identifier
 */
export const getColumns = (
  jurisdiction: GenericJurisdiction[],
  jurisdictionColumn: string,
  focusAreaColumn: string,
  focusAreaLevel: string,
  jurisdictionId: string | null
) => {
  const currentParent = jurisdiction.filter(el => el.jurisdiction_id === jurisdictionId);
  return currentParent &&
    currentParent.length > 0 &&
    currentParent[0].jurisdiction_depth >= +focusAreaLevel
    ? get(planTableColumnsLookup, focusAreaColumn, null)
    : get(planTableColumnsLookup, jurisdictionColumn, null);
};

/** Interface for linked cell props */
export interface LinkedCellProps extends DropDownCellProps {
  urlPath?: string;
}

/** Component that will be rendered in drop-down table cells showing a link
 * that moves you to the next hierarchical level.
 */
export const IRSMopupTableCell: React.ElementType<LinkedCellProps> = (props: LinkedCellProps) => {
  const { cell, cellValue, hasChildren, urlPath } = props;
  const original: Dictionary = cell.row.original;
  const url = urlPath ? `${urlPath}/${original.jurisdiction_id}` : '';
  return <div>{hasChildren ? <Link to={url}>{cellValue}</Link> : cellValue}</div>;
};
