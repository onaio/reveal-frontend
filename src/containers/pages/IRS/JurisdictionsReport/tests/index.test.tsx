import reducerRegistry from '@onaio/redux-reducer-registry';
import superset from '@onaio/superset-connector';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import flushPromises from 'flush-promises';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Helmet } from 'react-helmet';
import { Router } from 'react-router';
import { JurisdictionReport } from '../';
import { REPORT_IRS_PLAN_URL } from '../../../../../constants';
import store from '../../../../../store';
import GenericJurisdictionsReducer, {
  fetchGenericJurisdictions,
  GenericJurisdiction,
  getGenericJurisdictionByJurisdictionId,
  getGenericJurisdictionsArray,
  reducerName as GenericJurisdictionsReducerName,
} from '../../../../../store/ducks/generic/jurisdictions';
import IRSPlansReducer, {
  fetchIRSPlans,
  getIRSPlanById,
  IRSPlan,
  reducerName as IRSPlansReducerName,
} from '../../../../../store/ducks/generic/plans';
import genericStructuresReducer, {
  fetchGenericStructures,
  GenericStructure,
  getGenericStructures,
  reducerName as genericStructuresReducerName,
  StructureFeatureCollection,
} from '../../../../../store/ducks/generic/structures';
import { plans } from '../../../../../store/ducks/generic/tests/fixtures';
import jurisdictionReducer, {
  fetchJurisdictions,
  getJurisdictionById,
  Jurisdiction,
  reducerName as jurisdictionReducerName,
} from '../../../../../store/ducks/jurisdictions';
import * as fixtures from '../../JurisdictionsReport/tests/fixtures';
import { IRSTableColumns } from '../helpers';

/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

jest.mock('../../../../../configs/env');

/** register the reducers */
reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);
reducerRegistry.register(GenericJurisdictionsReducerName, GenericJurisdictionsReducer);

/** set up data in the store needed for this view */
const focusAreaData = superset.processData(fixtures.ZambiaFocusAreasJSON) || [];
const jurisdictionData = superset.processData(fixtures.ZambiaJurisdictionsJSON) || [];

store.dispatch(fetchGenericJurisdictions('zm-jurisdictions', jurisdictionData));
store.dispatch(fetchGenericJurisdictions('zm-focusAreas', focusAreaData));

const history = createBrowserHistory();

describe('components/IRS Reports/JurisdictionReport', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const mock: any = jest.fn();
    let jurisdictions =
      getGenericJurisdictionsArray(
        store.getState(),
        'zm-jurisdictions',
        '9f1e0cfa-5313-49ff-af2c-f7dbf4fbdb9d'
      ) || [];
    jurisdictions = jurisdictions.concat(
      getGenericJurisdictionsArray(
        store.getState(),
        'zm-focusAreas',
        '9f1e0cfa-5313-49ff-af2c-f7dbf4fbdb9d'
      )
    );
    const props = {
      history,
      jurisdictions,
      location: mock,
      match: {
        isExact: true,
        params: {
          planId: (plans[0] as IRSPlan).plan_id,
        },
        path: `${REPORT_IRS_PLAN_URL}/:planId`,
        url: `${REPORT_IRS_PLAN_URL}/${(plans[0] as IRSPlan).plan_id}`,
      },
      plan: plans[0] as IRSPlan,
    };
    shallow(
      <Router history={history}>
        <JurisdictionReport {...props} />
      </Router>
    );
  });

  it('renders correctly', async () => {
    fetch.mockResponseOnce(JSON.stringify({}));
    const mock: any = jest.fn();

    const supersetServiceMock: any = jest.fn();
    supersetServiceMock.mockImplementation(async () => []);

    let jurisdictions =
      getGenericJurisdictionsArray(
        store.getState(),
        'zm-jurisdictions',
        '9f1e0cfa-5313-49ff-af2c-f7dbf4fbdb9d'
      ) || [];
    jurisdictions = jurisdictions.concat(
      getGenericJurisdictionsArray(
        store.getState(),
        'zm-focusAreas',
        '9f1e0cfa-5313-49ff-af2c-f7dbf4fbdb9d'
      )
    );

    const props = {
      history,
      jurisdictions,
      location: mock,
      match: {
        isExact: true,
        params: {
          planId: (plans[0] as IRSPlan).plan_id,
        },
        path: `${REPORT_IRS_PLAN_URL}/:planId`,
        url: `${REPORT_IRS_PLAN_URL}/${(plans[0] as IRSPlan).plan_id}`,
      },
      plan: plans[0] as IRSPlan,
      service: supersetServiceMock,
    };

    const wrapper = mount(
      <Router history={history}>
        <JurisdictionReport {...props} />
      </Router>
    );
    const helmet = Helmet.peek();
    await flushPromises();
    expect(toJson(wrapper.find('BreadcrumbItem li'))).toMatchSnapshot('breadcrumbs');
    expect(toJson(wrapper.find('h3.page-title'))).toMatchSnapshot('page title');
    expect(helmet.title).toEqual('IRS Reporting: IRS 2019-09-05 TEST');
    expect(wrapper.find('DrillDownTable').props()).toMatchSnapshot({
      columns: expect.any(Object) /** just for purposes of making snapshot smaller */,
      data: expect.any(Object) /** just for purposes of making snapshot smaller */,
    });
    expect(wrapper.find('DrillDownTable').props().data).toEqual(jurisdictions);
    expect((wrapper.find('DrillDownTable').props() as any).columns).toEqual(
      IRSTableColumns.zambiaJurisdictions2019
    );
    expect(supersetServiceMock.mock.calls).toEqual([
      [
        '13',
        {
          adhoc_filters: [
            {
              clause: 'WHERE',
              comparator: '727c3d40-e118-564a-b231-aac633e6abce',
              expressionType: 'SIMPLE',
              operator: '==',
              subject: 'plan_id',
            },
          ],
          row_limit: 1,
        },
      ],
      [
        '11',
        {
          adhoc_filters: [
            {
              clause: 'WHERE',
              comparator: '727c3d40-e118-564a-b231-aac633e6abce',
              expressionType: 'SIMPLE',
              operator: '==',
              subject: 'plan_id',
            },
          ],
          order_by_cols: ['["jurisdiction_depth",+true]'],
          row_limit: 3000,
        },
      ],
      [
        '12',
        {
          adhoc_filters: [
            {
              clause: 'WHERE',
              comparator: '727c3d40-e118-564a-b231-aac633e6abce',
              expressionType: 'SIMPLE',
              operator: '==',
              subject: 'plan_id',
            },
          ],
          order_by_cols: ['["jurisdiction_depth",+true]'],
          row_limit: 3000,
        },
      ],
    ]);
    expect(supersetServiceMock).toHaveBeenCalledTimes(3);
    wrapper.unmount();
  });

  it('drills down correctly', async () => {
    fetch.mockResponseOnce(JSON.stringify({}));
    const mock: any = jest.fn();

    const supersetServiceMock: any = jest.fn();
    supersetServiceMock.mockImplementation(async () => []);

    let jurisdictions =
      getGenericJurisdictionsArray(
        store.getState(),
        'zm-jurisdictions',
        '9f1e0cfa-5313-49ff-af2c-f7dbf4fbdb9d'
      ) || [];
    jurisdictions = jurisdictions.concat(
      getGenericJurisdictionsArray(
        store.getState(),
        'zm-focusAreas',
        '9f1e0cfa-5313-49ff-af2c-f7dbf4fbdb9d'
      )
    );

    const props = {
      history,
      jurisdictions,
      location: mock,
      match: {
        isExact: true,
        params: {
          planId: (plans[0] as IRSPlan).plan_id,
        },
        path: `${REPORT_IRS_PLAN_URL}/:planId`,
        url: `${REPORT_IRS_PLAN_URL}/${(plans[0] as IRSPlan).plan_id}`,
      },
      plan: plans[0] as IRSPlan,
      service: supersetServiceMock,
    };

    const wrapper = mount(
      <Router history={history}>
        <JurisdictionReport {...props} />
      </Router>
    );

    const baseURL = `${REPORT_IRS_PLAN_URL}/727c3d40-e118-564a-b231-aac633e6abce`;

    expect(wrapper.find('HeaderBreadcrumb li').length).toEqual(3);
    expect(
      wrapper
        .find('HeaderBreadcrumb li')
        .first()
        .text()
    ).toEqual('Home');
    expect(
      wrapper
        .find('HeaderBreadcrumb a')
        .last()
        .text()
    ).toEqual('IRS Reporting');
    expect(
      wrapper
        .find('HeaderBreadcrumb li')
        .last()
        .text()
    ).toEqual('IRS 2019-09-05 TEST');
    expect(wrapper.find('HeaderBreadcrumb a').length).toEqual(2);
    expect(wrapper.find('h3.page-title').text()).toEqual('IRS Reporting: IRS 2019-09-05 TEST');
    expect(wrapper.find('DrillDownTable').props()).toMatchSnapshot({
      columns: expect.any(Object) /** just for purposes of making snapshot smaller */,
      data: expect.any(Object) /** just for purposes of making snapshot smaller */,
    });
    expect(wrapper.find('DrillDownTable').props().data).toEqual(jurisdictions);
    expect((wrapper.find('DrillDownTable').props() as any).columns).toEqual(
      IRSTableColumns.zambiaJurisdictions2019
    );

    // Zambia URL
    expect(
      wrapper.find(`a[href$="${baseURL}/22bc44dd-752d-4c20-8761-617361b4f1e7"]`).length
    ).toEqual(1);
    wrapper
      .find(`a[href$="${baseURL}/22bc44dd-752d-4c20-8761-617361b4f1e7"]`)
      .first()
      .simulate('click', { button: 0 });
    expect(wrapper.find('HeaderBreadcrumb li').length).toEqual(4);
    expect(wrapper.find('HeaderBreadcrumb a').length).toEqual(3);
    expect(
      wrapper
        .find('HeaderBreadcrumb a')
        .last()
        .text()
    ).toEqual('IRS 2019-09-05 TEST');
    expect(
      wrapper
        .find('HeaderBreadcrumb li')
        .last()
        .text()
    ).toEqual('Zambia');
    expect(wrapper.find('h3.page-title').text()).toEqual(
      'IRS Reporting: IRS 2019-09-05 TEST: Zambia'
    );
    expect(wrapper.find('DrillDownTable').props()).toMatchSnapshot({
      columns: expect.any(Object) /** just for purposes of making snapshot smaller */,
      data: expect.any(Object) /** just for purposes of making snapshot smaller */,
    });
    expect(wrapper.find('DrillDownTable').props().data).toEqual(jurisdictions);
    expect((wrapper.find('DrillDownTable').props() as any).columns).toEqual(
      IRSTableColumns.zambiaJurisdictions2019
    );

    // Lusaka URL
    expect(
      wrapper.find(`a[href$="${baseURL}/fd5a22ba-fa0e-4fdc-ae1c-1db8965b7b43"]`).length
    ).toEqual(1);
    wrapper
      .find(`a[href$="${baseURL}/fd5a22ba-fa0e-4fdc-ae1c-1db8965b7b43"]`)
      .first()
      .simulate('click', { button: 0 });
    expect(wrapper.find('HeaderBreadcrumb li').length).toEqual(5);
    expect(wrapper.find('HeaderBreadcrumb a').length).toEqual(4);
    expect(
      wrapper
        .find('HeaderBreadcrumb a')
        .last()
        .text()
    ).toEqual('Zambia');
    expect(
      wrapper
        .find('HeaderBreadcrumb li')
        .last()
        .text()
    ).toEqual('Lusaka');
    expect(wrapper.find('h3.page-title').text()).toEqual(
      'IRS Reporting: IRS 2019-09-05 TEST: Lusaka'
    );
    expect(wrapper.find('DrillDownTable').props()).toMatchSnapshot({
      columns: expect.any(Object) /** just for purposes of making snapshot smaller */,
      data: expect.any(Object) /** just for purposes of making snapshot smaller */,
    });
    expect(wrapper.find('DrillDownTable').props().data).toEqual(jurisdictions);
    expect((wrapper.find('DrillDownTable').props() as any).columns).toEqual(
      IRSTableColumns.zambiaJurisdictions2019
    );

    // Lusaka Test
    expect(
      wrapper.find(`a[href$="${baseURL}/8f1fd4cf-264d-4ed6-8838-e538c5845d46"]`).length
    ).toEqual(1);
    wrapper
      .find(`a[href$="${baseURL}/8f1fd4cf-264d-4ed6-8838-e538c5845d46"]`)
      .first()
      .simulate('click', { button: 0 });
    expect(wrapper.find('HeaderBreadcrumb li').length).toEqual(6);
    expect(wrapper.find('HeaderBreadcrumb a').length).toEqual(5);
    expect(
      wrapper
        .find('HeaderBreadcrumb a')
        .last()
        .text()
    ).toEqual('Lusaka');
    expect(
      wrapper
        .find('HeaderBreadcrumb li')
        .last()
        .text()
    ).toEqual('Lusaka_Test');
    expect(wrapper.find('h3.page-title').text()).toEqual(
      'IRS Reporting: IRS 2019-09-05 TEST: Lusaka_Test'
    );
    expect(wrapper.find('DrillDownTable').props()).toMatchSnapshot({
      columns: expect.any(Object) /** just for purposes of making snapshot smaller */,
      data: expect.any(Object) /** just for purposes of making snapshot smaller */,
    });
    expect(wrapper.find('DrillDownTable').props().data).toEqual(jurisdictions);
    expect((wrapper.find('DrillDownTable').props() as any).columns).toEqual(
      IRSTableColumns.zambiaJurisdictions2019
    );

    // Lusaka HFC
    expect(
      wrapper.find(`a[href$="${baseURL}/2bf9915d-8725-4061-983d-5938802ac0f0"]`).length
    ).toEqual(1);
    wrapper
      .find(`a[href$="${baseURL}/2bf9915d-8725-4061-983d-5938802ac0f0"]`)
      .first()
      .simulate('click', { button: 0 });
    expect(wrapper.find('HeaderBreadcrumb li').length).toEqual(7);
    expect(wrapper.find('HeaderBreadcrumb a').length).toEqual(6);
    expect(
      wrapper
        .find('HeaderBreadcrumb a')
        .last()
        .text()
    ).toEqual('Lusaka_Test');
    expect(
      wrapper
        .find('HeaderBreadcrumb li')
        .last()
        .text()
    ).toEqual('Lusaka HFC');
    expect(wrapper.find('h3.page-title').text()).toEqual(
      'IRS Reporting: IRS 2019-09-05 TEST: Lusaka HFC'
    );
    expect(wrapper.find('DrillDownTable').props()).toMatchSnapshot({
      columns: expect.any(Object) /** just for purposes of making snapshot smaller */,
      data: expect.any(Object) /** just for purposes of making snapshot smaller */,
    });
    expect(wrapper.find('DrillDownTable').props().data).toEqual(jurisdictions);
    expect((wrapper.find('DrillDownTable').props() as any).columns).toEqual(
      IRSTableColumns.zambiaFocusArea2019
    );

    expect(wrapper.find('DrillDownTable a').length).toEqual(14);
    expect(
      wrapper.find('DrillDownTable .plan-jurisdiction-name.main-span').map(e => e.text())
    ).toEqual([
      '  Akros 3',
      '  Akros_2',
      '  Ibex_1',
      '  Kalikiliki Settlement',
      '  Akros_1',
      '  Bauleni Settlement',
      '  Helen Kaunda Settlement',
    ]);
    expect(wrapper.find('DrillDownTable a').map(e => e.props().href)).toEqual([
      `${baseURL}/a049c269-6c06-47ef-b5b3-4f5f7736d97d/map`,
      `${baseURL}/a049c269-6c06-47ef-b5b3-4f5f7736d97d/map`,
      `${baseURL}/7aee968f-6b1f-427d-bba3-fa292937532b/map`,
      `${baseURL}/7aee968f-6b1f-427d-bba3-fa292937532b/map`,
      `${baseURL}/67f2211d-0cfc-4bde-be0e-bf8b5b3e0c2c/map`,
      `${baseURL}/67f2211d-0cfc-4bde-be0e-bf8b5b3e0c2c/map`,
      `${baseURL}/d7e42fb9-381c-4fc6-a4c1-7e112076c559/map`,
      `${baseURL}/d7e42fb9-381c-4fc6-a4c1-7e112076c559/map`,
      `${baseURL}/0dc2d15b-be1d-45d3-93d8-043a3a916f30/map`,
      `${baseURL}/0dc2d15b-be1d-45d3-93d8-043a3a916f30/map`,
      `${baseURL}/0b142aff-341c-4d15-878e-55942bc873aa/map`,
      `${baseURL}/0b142aff-341c-4d15-878e-55942bc873aa/map`,
      `${baseURL}/fdc73ce5-650a-4246-a2b3-a6665adc8249/map`,
      `${baseURL}/fdc73ce5-650a-4246-a2b3-a6665adc8249/map`,
    ]);
  });
});
