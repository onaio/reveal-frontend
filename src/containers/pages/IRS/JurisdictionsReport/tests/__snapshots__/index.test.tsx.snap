// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IRS Reports/JurisdictionReport drills down correctly 1`] = `
Object {
  "CellComponent": [Function],
  "columns": Any<Object>,
  "data": Any<Object>,
  "defaultPageSize": 154,
  "extraCellProps": Object {
    "urlPath": "/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce",
  },
  "getTdProps": [Function],
  "hasChildren": [Function],
  "identifierField": "jurisdiction_id",
  "linkerField": "jurisdiction_name",
  "minRows": 0,
  "parentIdentifierField": "jurisdiction_parent_id",
  "resizable": true,
  "rootParentId": "",
  "shouldUseEffect": false,
  "showPagination": false,
  "useDrillDownTrProps": false,
}
`;

exports[`components/IRS Reports/JurisdictionReport drills down correctly 2`] = `
Object {
  "CellComponent": [Function],
  "columns": Any<Object>,
  "data": Any<Object>,
  "defaultPageSize": 154,
  "extraCellProps": Object {
    "urlPath": "/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce",
  },
  "getTdProps": [Function],
  "hasChildren": [Function],
  "identifierField": "jurisdiction_id",
  "linkerField": "jurisdiction_name",
  "minRows": 0,
  "parentIdentifierField": "jurisdiction_parent_id",
  "resizable": true,
  "rootParentId": "22bc44dd-752d-4c20-8761-617361b4f1e7",
  "shouldUseEffect": false,
  "showPagination": false,
  "useDrillDownTrProps": false,
}
`;

exports[`components/IRS Reports/JurisdictionReport drills down correctly 3`] = `
Object {
  "CellComponent": [Function],
  "columns": Any<Object>,
  "data": Any<Object>,
  "defaultPageSize": 154,
  "extraCellProps": Object {
    "urlPath": "/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce",
  },
  "getTdProps": [Function],
  "hasChildren": [Function],
  "identifierField": "jurisdiction_id",
  "linkerField": "jurisdiction_name",
  "minRows": 0,
  "parentIdentifierField": "jurisdiction_parent_id",
  "resizable": true,
  "rootParentId": "fd5a22ba-fa0e-4fdc-ae1c-1db8965b7b43",
  "shouldUseEffect": false,
  "showPagination": false,
  "useDrillDownTrProps": false,
}
`;

exports[`components/IRS Reports/JurisdictionReport drills down correctly 4`] = `
Object {
  "CellComponent": [Function],
  "columns": Any<Object>,
  "data": Any<Object>,
  "defaultPageSize": 154,
  "extraCellProps": Object {
    "urlPath": "/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce",
  },
  "getTdProps": [Function],
  "hasChildren": [Function],
  "identifierField": "jurisdiction_id",
  "linkerField": "jurisdiction_name",
  "minRows": 0,
  "parentIdentifierField": "jurisdiction_parent_id",
  "resizable": true,
  "rootParentId": "8f1fd4cf-264d-4ed6-8838-e538c5845d46",
  "shouldUseEffect": false,
  "showPagination": false,
  "useDrillDownTrProps": false,
}
`;

exports[`components/IRS Reports/JurisdictionReport drills down correctly 5`] = `
Object {
  "CellComponent": [Function],
  "columns": Any<Object>,
  "data": Any<Object>,
  "defaultPageSize": 154,
  "extraCellProps": Object {
    "urlPath": "/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce",
  },
  "getTdProps": [Function],
  "hasChildren": [Function],
  "identifierField": "jurisdiction_id",
  "linkerField": "jurisdiction_name",
  "minRows": 0,
  "parentIdentifierField": "jurisdiction_parent_id",
  "resizable": true,
  "rootParentId": "2bf9915d-8725-4061-983d-5938802ac0f0",
  "shouldUseEffect": false,
  "showPagination": false,
  "useDrillDownTrProps": false,
}
`;

exports[`components/IRS Reports/JurisdictionReport renders correctly 1`] = `
Object {
  "CellComponent": [Function],
  "columns": Any<Object>,
  "data": Any<Object>,
  "defaultPageSize": 154,
  "extraCellProps": Object {
    "urlPath": "/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce",
  },
  "getTdProps": [Function],
  "hasChildren": [Function],
  "identifierField": "jurisdiction_id",
  "linkerField": "jurisdiction_name",
  "minRows": 0,
  "parentIdentifierField": "jurisdiction_parent_id",
  "resizable": true,
  "rootParentId": "",
  "shouldUseEffect": false,
  "showPagination": false,
  "useDrillDownTrProps": false,
}
`;

exports[`components/IRS Reports/JurisdictionReport renders correctly: breadcrumbs 1`] = `
Array [
  <li
    className="breadcrumb-item"
  >
    <Link
      key="0"
      replace={false}
      to="/"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="1"
      replace={false}
      to="/intervention/irs/report"
    >
      <a
        href="/intervention/irs/report"
        onClick={[Function]}
      >
        IRS Reporting
      </a>
    </Link>
  </li>,
  <li
    aria-current="page"
    className="active breadcrumb-item"
  >
    IRS 2019-09-05 TEST
  </li>,
]
`;

exports[`components/IRS Reports/JurisdictionReport renders correctly: page title 1`] = `
<h3
  className="mb-3 page-title"
>
  IRS Reporting: IRS 2019-09-05 TEST
</h3>
`;
